src/Csv.php

```diff
@@ -444,14 +444,28 @@ public function output($filename = null, $data = array(), $fields = array(), $de
             $mime = $delimiter === "\t" ?
                 'text/tab-separated-values' :
                 'application/csv';
+
+            // Buffer output:
+            ob_start();
             header('Content-type: ' . $mime);
-            header('Content-Length: ' . strlen($flat_string));
             header('Cache-Control: no-cache, must-revalidate');
             header('Pragma: no-cache');
             header('Expires: 0');
             header('Content-Disposition: attachment; filename="' . $filename . '"; modification-date="' . date('r') . '";');
 
-            echo $flat_string;
+            // gzip enabled?
+            if (isset($_SERVER['HTTP_ACCEPT_ENCODING']) 
+                && stripos($_SERVER['HTTP_ACCEPT_ENCODING'], "gzip") !== false) {
+                ob_start("ob_gzhandler");
```

Comment:
1. We also need to check if the `zlib` extension is enabled, e.g. by calling `function_exists('ob_gzhandler')`.
2. We need to tell the browser that we compressed: ``header("Content-Encoding: gzip");``